device/Cargo.toml

```diff
@@ -16,23 +16,26 @@ mem_debug = []
 [dependencies]
 frostsnap_core = { workspace = true }
 frostsnap_comms = { workspace = true }
-esp-hal = { package = "esp-hal", version = "0.16.1", features = ["esp32c3"] }
+esp-hal = { package = "esp-hal", version = "0.18.0", features = ["esp32c3"] }
 esp-alloc = "0.3.0"
 nb = "1"
+embedded-hal-nb = "1"
+embedded-hal-bus = "0.1.0"
 embedded-storage = "0.3.1"
 esp-storage = { version = "0.3.0", features = ["esp32c3"] }
 bincode = { workspace = true }
-display-interface = { version = "0.4.1", optional = true }
-display-interface-spi = { version = "0.4.1", optional = true }
-mipidsi = { version = "0.7.1", features = ["batch"], optional = true }
+display-interface = { version = "0.5.0", optional = true }
+display-interface-spi = { version = "0.5.0", optional = true }
+mipidsi = { version = "0.8.0", features = ["batch"], optional = true }
 embedded-graphics-framebuf = { version = "0.5.0", optional = true }
 embedded-graphics = "0.8.1"
 embedded-text = "0.7.0"
 u8g2-fonts = { version = "0.3.0", features = ["embedded_graphics_textstyle"] }
 rand_chacha = { workspace = true }
-cst816s = "0.1.4"
-embedded-hal = { version = "0.2.7" }
+cst816s = { git = "https://github.com/fxweidinger/cst816s", rev = "82f88186e94a94126b962d30b4c0129fd3c1c5f8" }
```

Comment:
We should probably vendor this inside the workspace at some point if we're going to use it in production.